coderd/database/dbauthz/dbauthz.go

```diff
@@ -1992,6 +1992,10 @@ func (q *querier) GetLogoURL(ctx context.Context) (string, error) {
 	return q.db.GetLogoURL(ctx)
 }
 
+func (q *querier) GetNotStartedProvisionerJobs(ctx context.Context, notStartedSince time.Time) ([]database.ProvisionerJob, error) {
```

Comment:
Needs an authz check. We don't have a rbac resource type for provisioner jobs unfortunately so maybe you should just assert that the subject can `update` the `system` resource unless there's something better you could use instead

Reply:
We actually do now: https://github.com/coder/coder/blob/main/coderd/rbac/object_gen.go#L235-L240

You might need to add `ActionUpdate` as an available action on them here: 

https://github.com/coder/coder/blob/main/coderd/rbac/policy/policy.go#L183-L187

And then `subjectReaper`  gets the ability to update all provisioner jobs at site level.